pkg/plugins/mellanox/mellanox_plugin.go

```diff
@@ -206,6 +206,29 @@ func (p *MellanoxPlugin) Apply() error {
 		return err
 	}
 	if vars.FeatureGate.IsEnabled(consts.MellanoxFirmwareResetFeatureGate) {
+		for _, pciAddress := range pciAddressesToReset {
+			err := p.helpers.SetSriovNumVfs(pciAddress, 0)
+			if err != nil {
+				log.Log.Error(err, "failed to set SR-IOV number of VFs to 0 before firmware reset", "pciAddress", pciAddress)
+				return err
+			}
+
+			if mlx.IsDualPort(pciAddress, mellanoxNicsStatus) {
+				otherIfaceSpec := mlx.GetOtherPortSpec(pciAddress, mellanoxNicsSpec)
```

Comment:
we would want to reset the VFs on the second PF regardless of it being specified in spec, else the reset command fails no ?